enterprise/cypher/cypher-compiled-runtime-3.2/src/main/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/codegen/LogicalPlanConverter.scala

```diff
@@ -526,6 +526,10 @@ object LogicalPlanConverter {
         case CypherCodeGenType(symbols.ListType(innerType), ListReferenceType(innerReprType))
           if RepresentationType.isPrimitive(innerReprType) =>
           (CypherCodeGenType(innerType, innerReprType), UnwindPrimitiveCollection(opName, collection))
+        case CypherCodeGenType(symbols.ListType(innerType), ListReferenceType(innerReprType)) =>
+          (CypherCodeGenType(innerType, innerReprType), ir.UnwindCollection(opName, collection))
+        case CypherCodeGenType(symbols.ListType(innerType), _) =>
```

Comment:
Now that case is duplicate... that is probably not intentional?

Reply:
No not quite, in the case where the inner type is primitive we use `UnwindPrimitiveCollection` and in the other we use ` ir.UnwindCollection`

Reply:
Sorry, I should have been more specific. Lines 531 and 533.

Reply:
yes you are right, that was a mistake :)